state/stateChanges/collector.go

```diff
@@ -1,311 +0,0 @@
-}
-
-// Publish will export state changes
-func (c *collector) Publish() (map[string]*data.StateChanges, error) {
-	c.stateChangesMut.RLock()
-	defer c.stateChangesMut.RUnlock()
-
-	stateChangesForTxs := make(map[string]*data.StateChanges)
-	for _, stateChange := range c.stateChanges {
-		txHash := string(stateChange.GetTxHash())
-		if len(txHash) == 0 {
-			log.Warn("empty tx hash, state change event not associated to a transaction", "stateChange", stateChangeToString(stateChange))
-			continue
-		}
-
-		st, ok := stateChange.(*data.StateChange)
-		if !ok {
-			log.Warn("failed to cast state change to data.StateChange", "stateChange", stateChangeToString(stateChange))
-			continue
-		}
-
-		_, ok = stateChangesForTxs[txHash]
-		if !ok {
-			log.Trace("created new state changes for tx", "txHash", txHash)
-			stateChangesForTxs[txHash] = &data.StateChanges{
-				StateChanges: []*data.StateChange{st},
-			}
-		} else {
-			log.Trace("appended state change to existing state changes for tx", "txHash", txHash)
-			stateChangesForTxs[txHash].StateChanges = append(stateChangesForTxs[txHash].StateChanges, st)
-		}
-	}
-	log.Trace("published state changes", "numTxs", len(stateChangesForTxs))
-
-	return stateChangesForTxs, nil
-}
-
-func stateChangeToString(stateChange state.StateChange) string {
-	dataTrieChanges := make([]string, len(stateChange.GetDataTrieChanges()))
-	for i, dataTrieChange := range stateChange.GetDataTrieChanges() {
-		dataTrieChanges[i] = fmt.Sprintf("key: %v, val: %v, type: %v", hex.EncodeToString(dataTrieChange.Key), hex.EncodeToString(dataTrieChange.Val), dataTrieChange.Type)
-	}
-	return fmt.Sprintf("type: %v, operation: %v, mainTrieKey: %v, mainTrieVal: %v, index: %v, dataTrieChanges: %v",
-		stateChange.GetType(),
-		stateChange.GetOperation(),
-		hex.EncodeToString(stateChange.GetMainTrieKey()),
-		hex.EncodeToString(stateChange.GetMainTrieVal()),
-		stateChange.GetIndex(),
-		strings.Join(dataTrieChanges, ", "),
-	)
-}
-
-// Store will store the collected state changes if it has been configured with a storer
-func (c *collector) Store() error {
-	// TODO: evaluate adding a more explicit field check here
-	if check.IfNil(c.storer) {
-		return nil
-	}
-
-	stateChangesForTx, err := c.getDataAnalysisStateChangesForTxs()
-	if err != nil {
-		return fmt.Errorf("failed to retrieve data analysis state changes for tx: %w", err)
-	}
-
-	for _, stateChange := range stateChangesForTx {
-		marshalledData, err := json.Marshal(stateChange)
-		if err != nil {
-			return fmt.Errorf("failed to marshal state changes to JSON: %w", err)
-		}
-
-		err = c.storer.Put(stateChange.TxHash, marshalledData)
-		if err != nil {
-			return fmt.Errorf("failed to store marshalled data: %w", err)
-		}
-	}
-
-	return nil
-}
-
-// AddTxHashToCollectedStateChanges will try to set txHash field to each state change
-// if the field is not already set
-func (c *collector) AddTxHashToCollectedStateChanges(txHash []byte, tx coreData.TransactionHandler) {
```

Comment:
we still need some tx fields for data analysis, like sender, receiver, data, so we have to gather that info for data analysis flow

Reply:
We can get that data from the txPool or from indexer or from somewhere else. There is no point in also saving the tx with the state accesses.